lightning_client.go

```diff
@@ -2703,3 +2708,57 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// InvoiceSubscription holds the parameters to specify from which update
+// to start streaming.
+type InvoiceSubscription struct {
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all added indexes with an add_index greater than this
+	//value. This allows callers to catch up on any events they missed while they
+	//weren't connected to the streaming RPC.
+	AddIndex uint64
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all settled indexes with an settle_index greater than
+	//this value. This allows callers to catch up on any events they missed while
+	//they weren't connected to the streaming RPC.
+	SettleIndex uint64
+}
+
+// SubscribeInvoices allows a client to subscribe to updates
+// of newly added/settled invoices.
+func (s *lightningClient) SubscribeInvoices(ctx context.Context,
+	req InvoiceSubscription) (<-chan *lnrpc.Invoice, <-chan error, error) {
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+	invoiceStream, err := s.client.SubscribeInvoices(rpcCtx,
```

Comment:
nit: preferred formatting for this is:

```
	invoiceStream, err := s.client.SubscribeInvoices(
		rpcCtx, &lnrpc.InvoiceSubscription{
			AddIndex:    req.AddIndex,
			SettleIndex: req.SettleIndex,
		},
	)
```

Reply:
Go fmt formats back to:
`
	invoiceStream, err := s.client.SubscribeInvoices(
		rpcCtx, &lnrpc.InvoiceSubscription{
			AddIndex:    req.AddIndex,
			SettleIndex: req.SettleIndex,
		})
`
is it ok?

Reply:
Strange, even if you add the comma after the closing `}`? But yeah, if it changes it back, then leave it.

Reply:
I tried with the comma, I still have the same issue "/. Let me know if I should change something.